### PR TITLE
Chore: camera reel behaviour critical fixes

### PR DESCRIPTION
**Pull request by popuz**
## What does this PR change?

- handled overload exceptions
- add PlaceId when uploading sceenshot
- add cooldown to prevent screencapture spamming

## How to test the changes?

1. Launch the explorer
2. Verify you that you cannot make more than 500 images and and that uploading last (500th) image is handled properly - not present any issues.
3. Verify you cannot spam screenshoting

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: Explorer/Assets/DCL/InWorldCamera/InWorldCamera/ScreenshotLimitReachedException.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿using System;
+
+namespace DCL.InWorldCamera.Systems
+{
+    public class ScreenshotLimitReachedException : Exception { }
```
Comment (AnsisMalins) [inline Explorer/Assets/DCL/InWorldCamera/InWorldCamera/ScreenshotLimitReachedException.cs:5]: The class can be `sealed`.

File: Explorer/Assets/DCL/InWorldCamera/InWorldCamera/ScreenshotMetadataBuilder.cs [hunk 2]
```diff
@@ -97,7 +97,8 @@ private static string[] FilterNonBaseWearables(IReadOnlyCollection<URN> avatarWe
             return wearables.ToArray();
         }
 
-        internal void FillMetadata(Profile profile, RealmData realm, Vector2Int playerPosition, string sceneName, VisiblePerson[] visiblePeople)
+        internal void FillMetadata(Profile profile, RealmData realm, Vector2Int playerPosition, string sceneName, string placeId,
+            VisiblePerson[] visiblePeople)
```
Comment (AnsisMalins) [inline Explorer/Assets/DCL/InWorldCamera/InWorldCamera/ScreenshotMetadataBuilder.cs:101]: Consider observing a line length that fits into Github's UI. I've measured it to be 105 characters.
image

File: Explorer/Assets/DCL/InWorldCamera/InWorldCamera/UI/InWorldCameraView.cs [hunk 3]
```diff
@@ -96,6 +94,10 @@ private Tween AnimateVFXImageTransition(float duration)
         private Tween AnimateVFXImageScale(float duration) =>
             animatedImage.rectTransform.DOScale(Vector2.zero, duration)
                          .SetEase(Ease.InOutQuad)
-                         .OnComplete(() => { animatedImage.rectTransform.localScale = Vector2.one; });
+                         .OnComplete(() =>
+                          {
+                              animatedImage.enabled = false;
+                              animatedImage.rectTransform.localScale = Vector2.one;
```
Comment (AnsisMalins) [inline Explorer/Assets/DCL/InWorldCamera/InWorldCamera/UI/InWorldCameraView.cs:100]: RectTransform.localScale is of type Vector3.

**Review by AnsisMalins — APPROVED** [review]

**Review by lorenzo-ranciaffi — APPROVED** [review]

File: Explorer/Assets/DCL/InWorldCamera/InWorldCamera/ScreenshotMetadataBuilder.cs (hunk 2 above)
Comment (popuz) [reply Explorer/Assets/DCL/InWorldCamera/InWorldCamera/ScreenshotMetadataBuilder.cs:101]: we can agree with the team on this line break value and put it in the formatting. On the time when we were agreeing, team decided to not enforce line length break

**Review by DafGreco — APPROVED** [review]
Approving this PR since you are not able to take more than 500 photos. 

The screenshot UI appears after the 500th photo but does not seem to take the photo itself.

**Comment by AnsisMalins** [comment]
That should be fixed. The UI should not deceive the user.

**Comment by popuz** [comment]
Ok. I will handle UI part in the another PR then

**Pull request merged**